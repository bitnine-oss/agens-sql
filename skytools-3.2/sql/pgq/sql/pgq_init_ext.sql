
-- create noext schema
\set ECHO none
\set VERBOSITY 'terse'
set client_min_messages = 'warning';
\i structure/install.sql
select pgq.create_queue('testqueue1');
\set ECHO all
-- convert to extension
create extension pgq from 'unpackaged';
select array_length(extconfig, 1) from pg_catalog.pg_extension where extname = 'pgq';

select pgq.create_queue('testqueue2');
--drop extension pgq; -- will fail
select pgq.drop_queue('testqueue2');
select pgq.drop_queue('testqueue1');

-- drop schema failure
drop extension pgq;

-- create clean schema
create extension pgq;

select array_length(extconfig, 1) from pg_catalog.pg_extension where extname = 'pgq';

