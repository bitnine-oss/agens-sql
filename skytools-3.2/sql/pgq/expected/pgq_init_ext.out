-- create noext schema
\set ECHO none
 upgrade_schema 
----------------
              0
(1 row)

 create_queue 
--------------
            1
(1 row)

-- convert to extension
create extension pgq from 'unpackaged';
select array_length(extconfig, 1) from pg_catalog.pg_extension where extname = 'pgq';
 array_length 
--------------
            7
(1 row)

select pgq.create_queue('testqueue2');
 create_queue 
--------------
            1
(1 row)

--drop extension pgq; -- will fail
select pgq.drop_queue('testqueue2');
 drop_queue 
------------
          1
(1 row)

select pgq.drop_queue('testqueue1');
 drop_queue 
------------
          1
(1 row)

-- drop schema failure
drop extension pgq;
-- create clean schema
create extension pgq;
select array_length(extconfig, 1) from pg_catalog.pg_extension where extname = 'pgq';
 array_length 
--------------
            7
(1 row)

